server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/configuration/ConfigurationAggregate.java

```diff
@@ -0,0 +1,124 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.queue.rabbitmq.view.cassandra.configuration;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.eventsourcing.AggregateId;
+import org.apache.james.eventsourcing.Event;
+import org.apache.james.eventsourcing.eventstore.History;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+
+class ConfigurationAggregate {
+
+    private static class State {
+
+        private static final Logger LOGGER = LoggerFactory.getLogger(State.class);
+
+        private static State initial() {
+            return new State(Optional.empty());
+        }
+
+        private Optional<CassandraMailQueueViewConfiguration> maybeConfiguration;
+
+        State(Optional<CassandraMailQueueViewConfiguration> maybeConfiguration) {
+            Preconditions.checkNotNull(maybeConfiguration);
+            this.maybeConfiguration = maybeConfiguration;
+        }
+
+        boolean set(CassandraMailQueueViewConfiguration configuration) {
+            validateConfiguration(configuration);
+            boolean isSame = maybeConfiguration
+                .map(currentConfiguration -> currentConfiguration.equals(configuration))
+                .orElse(false);
+
+            if (!isSame) {
```

Comment:
Good goal but that is not enough.

Here, in case you save two time the same config, you dio update the state one time, but nothing prevent to store a duplicated event (!)

That safety should be in `applyConfiguration`, which should return an empty list of event when we have the same config. BTW we do not really care about avoiding noop state updates.